### PR TITLE
add importpath_aliases to go_proto_library

### LINKED ISSUE #2358: importpath_aliases not implemented for go_proto_library
### What version of rules_go are you using?

v0.21.2

### What version of gazelle are you using?

v0.19.1

### What version of Bazel are you using?

2.0.0

### Does this issue reproduce with the latest releases of all the above?

Yes

### What operating system and processor architecture are you using?

Ubuntu amd64

### Any other potentially useful information about your toolchain?

N/A

### What did you do?

blaze run //:gazelle -- update-repos github.com/dgraph-io/badger/v2

(v2 isn't a real directory but an module named in:  https://github.com/dgraph-io/badger/blob/master/go.mod)

### What did you expect to see?

The result should build

### What did you see instead?

$ blaze build :all
ERROR: /home/mike/.cache/bazel/_bazel_mike/00f1347297fd781be7971b81296fd31c/external/com_github_dgraph_io_badger_v2/pb/BUILD.bazel:10:1: @com_github_dgraph_io_badger_v2//pb:pb_go_proto: no such attribute 'importpath_aliases' in 'go_proto_library' rule
ERROR: /home/mike/.cache/bazel/_bazel_mike/00f1347297fd781be7971b81296fd31c/external/com_github_dgraph_io_badger_v2/pb/BUILD.bazel:4:1: Target '@com_github_dgraph_io_badger_v2//pb:pb.proto' contains an error and its package is in error and referenced by '@com_github_dgraph_io_badger_v2//pb:pb_proto'
ERROR: /home/mike/.cache/bazel/_bazel_mike/00f1347297fd781be7971b81296fd31c/external/com_github_dgraph_io_badger_v2/pb/BUILD.bazel:10:1: Target '@com_github_dgraph_io_badger_v2//pb:pb_proto' contains an error and its package is in error and referenced by '@com_github_dgraph_io_badger_v2//pb:pb_go_proto'
ERROR: /home/mike/.cache/bazel/_bazel_mike/00f1347297fd781be7971b81296fd31c/external/com_github_dgraph_io_badger_v2/pb/BUILD.bazel:18:1: Target '@com_github_dgraph_io_badger_v2//pb:pb_go_proto' contains an error and its package is in error and referenced by '@com_github_dgraph_io_badger_v2//pb:go_default_library'
ERROR: /home/mike/.cache/bazel/_bazel_mike/00f1347297fd781be7971b81296fd31c/external/com_github_dgraph_io_badger_v2/BUILD.bazel:3:1: Target '@com_github_dgraph_io_badger_v2//pb:go_default_library' contains an error and its package is in error and referenced by '@com_github_dgraph_io_badger_v2//:go_default_library'
ERROR: Analysis of target '//collecty:go_default_library' failed; build aborted: Analysis failed
INFO: Elapsed time: 1.602s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (3 packages loaded, 203 targets configured)

-------

I see importpath_aliases is defined for go_library, but not go_proto_library?  However gazelle generates BUILD files that use them:

go_proto_library(
    name = "pb_go_proto",
    importpath = "github.com/dgraph-io/badger/v2/pb",
    importpath_aliases = ["github.com/dgraph-io/badger/pb"],
    proto = ":pb_proto",
    visibility = ["//visibility:public"],
)

### PR DESCRIPTION
**Pull request by tian000**
**What type of PR is this?**
Bug fix

**What does this PR do? Why is it needed?**
It allows go_proto_library to accept `importpath_aliases` as an argument. It is needed because gazelle generates go_proto_library rules that have importpath_aliases as an argument.

**Which issues(s) does this PR fix?**
Fixes #2358

### CODE REVIEW

**Comment by jayconrod** [comment]
This PR seems incomplete? It doesn't actually implement `importpath_aliases` or test that it works.

**Comment by tian000** [comment]
@jayconrod in Line 19 of `tests/core/go_proto_library/protos_alias_test.go`, I import the protos using the alias path, and then ensure that the generated types exist.

**Comment by tian000** [comment]
Im pretty new to rules_go but i think it works out of the box because `go_context` grabs the attribute here:
https://github.com/ConsultingMD/rules_go/blob/3b5f0c950e0deec63cd77cdcec7765b5f41fefbd/go/private/context.bzl#L441

File: tests/core/go_proto_library/protos_alias_test.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/* Copyright 2018 The Bazel Authors. All rights reserved.
```
Comment (jayconrod) [inline tests/core/go_proto_library/protos_alias_test.go:1]: Update to 2020

File: tests/core/go_proto_library/BUILD.bazel [hunk 2]
```diff
@@ -117,9 +117,16 @@ go_test(
     deps = [":protos_go_proto"],
 )
 
+go_test(
```
Comment (jayconrod) [inline tests/core/go_proto_library/BUILD.bazel:120]: Mention this in README.rst with a brief explanation.

File: tests/core/go_proto_library/protos_alias_test.go (hunk 1 above)
Comment (tian000) [reply tests/core/go_proto_library/protos_alias_test.go]: done!

File: tests/core/go_proto_library/BUILD.bazel (hunk 2 above)
Comment (tian000) [reply tests/core/go_proto_library/BUILD.bazel:120]: done!

**Comment by tian000** [comment]
np! @jayconrod responded to your comments!

**Pull request merged**